torch/_dynamo/variables/user_defined.py

```diff
@@ -329,14 +329,21 @@ def call_function(
             and hasattr(
                 self.value, "__exit__"
             )  # TODO(voz): These can invoke user code!
-            and check_constant_args(args, kwargs)
-            and self.value.__init__ == object.__init__
+            # and check_constant_args(args, kwargs)
+            # and self.value.__init__ == object.__init__
```

Comment:
The `self.value.__init__ == object.__init__` was here to avoid user code to be executed in the case of failure ([ref](https://github.com/pytorch/pytorch/pull/117818))